### PR TITLE
MVP CLI Proposal implementation: leaf and root interactions

### LINKED ISSUE #157: MVP CLI Proposal implementation: leaf and root interactions
Continuation of implementation of issue #70

- [ ] CLI command to set the next leaf
- [ ] CLI command to delete the leaf
- [ ] CLI command to get the root to the file

### PR DESCRIPTION
**Pull request by tyshko-rostyslav**
Closes #157

### CODE REVIEW

File: rln-cli/src/main.rs [hunk 1]
```diff
@@ -82,6 +82,29 @@ fn main() -> Result<()> {
...
+        }
+        Some(Commands::DeleteLeaf { index }) => {
+            state
+                .rln
+                .ok_or(Report::msg("no RLN initialized"))?
+                .delete_leaf(*index)?;
+            Ok(())
+        }
+        Some(Commands::GetRoot { file }) => {
+            let output_data = File::open(&file)?;
```
Comment (rymnc) [inline rln-cli/src/main.rs:101]: Can we just log to stdout here?

**Review by rahulghangas — CHANGES_REQUESTED** [review]

File: rln-cli/src/main.rs [hunk 2]
```diff
@@ -82,6 +82,29 @@ fn main() -> Result<()> {
                 .init_tree_with_leaves(input_data)?;
             Ok(())
         }
+        Some(Commands::SetNextLeaf { file }) => {
+            let input_data = File::open(&file)?;
+            state
+                .rln
+                .ok_or(Report::msg("no RLN initialized"))?
```
[the remark this answers is not in the document]
Comment (tyshko-rostyslav) [reply rln-cli/src/main.rs]: Fixed

File: rln-cli/src/main.rs [hunk 3]
```diff
@@ -82,6 +82,29 @@ fn main() -> Result<()> {
...
+                .set_next_leaf(input_data)?;
+            Ok(())
+        }
+        Some(Commands::DeleteLeaf { index }) => {
+            state
+                .rln
+                .ok_or(Report::msg("no RLN initialized"))?
```
[the remark this answers is not in the document]
Comment (tyshko-rostyslav) [reply rln-cli/src/main.rs]: Fixed

File: rln-cli/src/main.rs [hunk 4]
```diff
@@ -82,6 +82,29 @@ fn main() -> Result<()> {
...
+                .delete_leaf(*index)?;
+            Ok(())
+        }
+        Some(Commands::GetRoot { file }) => {
+            let output_data = File::open(&file)?;
+            state
+                .rln
+                .ok_or(Report::msg("no RLN initialized"))?
```
[the remark this answers is not in the document]
Comment (tyshko-rostyslav) [reply rln-cli/src/main.rs]: Fixed

File: rln-cli/src/main.rs (hunk 1 above)
Comment (tyshko-rostyslav) [reply rln-cli/src/main.rs]: Redirected the output

**Review by rymnc — APPROVED** [review]
LGTM, thanks
for the future prs, let all the outputs be directed to stdout
we should also consider working on documentation for it too.

**Review by curryrasul — APPROVED** [review]
LGTM too!

**Review by rahulghangas — APPROVED** [review]

**Pull request merged**